### ISSUE TITLE
zstd does not follow quiet and input-removal conventions by default

### ISSUE THREAD

**Issue by chungy** [facebook/zstd#1442]
Precedence by pretty much all standard Unix compressors in the past, eg `compress`, `gzip`, `xz`, all follow the same behavior by default, that being non-verbose output and the removal of input files upon a successful compression or decompression. `zstd` does not follow the convention, giving rather surprising results. The standard mode for programs is pretty well-described in [_The Art of Unix Programming_](http://www.catb.org/esr/writings/taoup/html/ch01s06.html).

While somewhat rectifiable by aliasing `zstd` to `zstd -q --rm`, I think the opposite really ought to be the normal mode. The `-k` and `-v` switches that likewise are part of `gzip` and `xz` are already in the program, and I think it would be fine to keep the other two as well in case of overriding aliases.

**Comment by Cyan4973** [comment]
This topic has [already](https://github.com/facebook/zstd/pull/919) been [discussed](https://github.com/facebook/zstd/issues/1132).

tldr; `zstd` cli values human interaction, and especially the first user contact, while still allowing classical behavior through switches for integration into existing scripts.

This is a difference compared to previous generation utilities, which valued script integration over human interaction (mostly by virtue of copying existing behavior from past utilities).

The topic was on discussion up to v1.0, but has been settled since.
We need to retain it for user base consistency.
Generally speaking, I'm confident that the new default behavior is more in line with 21st century expectations.

Advanced users which prefer some different default behavior can indeed change it through `alias`, as you already mentioned.

Another route, currently considered, is to allow a set of environment variables to modify default behavior. A related issue is [opened here](https://github.com/facebook/zstd/issues/1423).
It needs to be done differently than `$GZIP` and `$XZ_OPT` though, as they are too opened and ended up opening security hazards. But a set of dedicated environment variables with very limited roles could avoid these security issues, while offering interesting customizations.

**Comment by grothesque** [comment]
Zstd is great, but I agree with @chungy  that the decision not to make `-q --rm` the default is unfortunate.

If the user wishes to compress some files on the command line, the originals typically serve no purpose, and can be recreated whenever needed.  The user interface of compress and similar tools gets this job done with the least fuzz and effort.

Moreover, in the common use case where one does not need the originals, having to delete them manually is actually less safe, since it requires explicit use of `rm` (or similar tool), which is potentially error-prone.

I understand that changing defaults is problematic, but what about introducing alternative commands, e.g. `zz` or `zs` that follow the traditional behavior, as discussed in
https://github.com/facebook/zstd/pull/919#issuecomment-348849187?

**Comment by grothesque** [comment]
A more "philosophical" take on this topic:

Unix has a unique convention for compressed *individual* files (e.g. `README.txt.xz`, `syslog.2.gz`).  This is notably different from archive formats like tar, zip or rar that combine (and optionally compress) whole filesystem trees.

The essence of this convention is that the compressed files almost become work-alikes of the originals.  There is a whole "ecosystem" of tools that serves this purpose:
- There’s `zcat`, `zless`, `zgrep`, etc.
- Many other tools accept and produce compressed files transparently.  For example, in the Emacs editor one can directly open and save gz- or xz-compressed files.
- To give another example, many different programs can work with svgz files.
- Compressing and uncompressing individual files is facilitated by simple, dedicated commands, e.g. `gzip` and `gunzip`.

All this allows to easily mix compressed and uncompressed files.  For example, one can `zgrep` the contents of `/var/log`.

Part of this convention is that compressing/uncompressing deletes the originals (and is silent on success).  I argued in the comment above that this behavior is useful and safer, but most importantly, it is part of a very well established convention.

It is confusing for a tool to fit into an established convention, but violate a part of it in a surprising (and arguably not useful) way.

Of course an individual user can set up aliases, but defaults do matter.  The current default behavior of zstd risks watering down a useful convention that has served us well for almost 40 years.

**Comment by gcflymoto** [comment]
I'll add that as part of automation flows, CI/CD pipelines, scripting, and so on, there is almost never a need to keep the originals. Tools like the very convenient [zutils](https://www.nongnu.org/zutils/zutils.html) make working with compressed files very transparent.

Another option is when TERM is defined zstandard is verbose, when it is undefined it is quiet.
